modules/fedimint-wallet-client/src/client_db.rs

```diff
@@ -14,11 +20,82 @@ impl std::fmt::Display for DbKeyPrefix {
     }
 }
 
+/// An index of a deposit address
+///
+/// Under the hood it's similar to `ChildId`, but in a wallet module
+/// it's used often enough to deserve own newtype.
+#[derive(Copy, Clone, Debug, Encodable, Decodable, Serialize, Default)]
+pub struct TweakIdx(pub u64);
+
+impl fmt::Display for TweakIdx {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_fmt(format_args!("TweakIdx({})", self.0))
+    }
+}
+
+impl TweakIdx {
+    #[must_use]
+    pub fn next(self) -> Self {
+        Self(self.0 + 1)
+    }
+}
+
+/// A counter tracking next index to use to derive a peg-in address
 #[derive(Clone, Debug, Encodable, Decodable, Serialize)]
 pub struct NextPegInTweakIndexKey;
 
 impl_db_record!(
     key = NextPegInTweakIndexKey,
-    value = u64,
+    value = TweakIdx,
```

Comment:
Does this require a client db migration?

Reply:
i believe "Encodable, Decodable" always means yes?